framework/src/bundle/BundleRegistry.cpp

```diff
@@ -249,8 +249,10 @@ namespace cppmicroservices
                 {
                     // Wait for the install thread to notify this thread that it is safe
                     // to install the current bundle
-                    auto& p = initialBundleInstallMap[location];
                     l.UnLock();
+#ifdef US_ENABLE_THREADING_SUPPORT
```

Comment:
the `#ifdef` here changes the behavior in the case where threading is enabled, `l.UnLock()` is in the wrong place.

Reply:
thinking about this more and digging into the problem, it looks like the gnu C++ library on the github build machines we use does not run into the linker problem we see on Debian 11:

```
CMakeFiles/CppMicroServices.dir/src/bundle/BundleRegistry.cpp.o: in function `std::cv_status std::condition_variable::__wait_until_impl<std::chrono::duration<long, std::ratio<1l, 1000000000l> > >(std::unique_lock<std::mutex>&, std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > > const&)':
/usr/include/c++/10/condition_variable:209: undefined reference to `pthread_cond_clockwait'
```

This PR has nothing to do with the abovementioned linker error, so I would prefer to treat this failure separately in another pull request. The fact that BundleRegistry.cpp uses condition_variables and mutexes means that it should be properly guarded when building without threading support, which is currently does not.

To be honest, I would prefer to eliminate the "no threading" build configuration entirely however that's a larger change with an unknown level of downstream impact that I don't currently have the bandwidth to address.

Reply:
The changes have been reverted